Sources/Rendering/Core/CellPicker/index.js

```diff
@@ -198,14 +199,31 @@ function vtkCellPicker(publicAPI, model) {
         model.pCoords = pickData.pCoords;
       }
     } else if (mapper.isA('vtkVolumeMapper')) {
-      tMin = publicAPI.intersectVolumeWithLine(
+      // we calculate here the parametric intercept points between the ray and the bounding box, so
+      // if the application defines for some reason a too large ray length (1e6), it restrict the calculation
+      // to the vtkVolume actor bounding box
+      const interceptionObject = vtkBox.intersectWithLine(
+        mapper.getBounds(),
         p1,
-        p2,
-        clipLine.t1,
-        clipLine.t2,
-        tol,
-        actor
+        p2
       );
+
+      if (interceptionObject) {
+        t1 =
+          interceptionObject.t1 < clipLine.t1
+            ? clipLine.t1
+            : interceptionObject.t1;
+
+        t2 =
+          interceptionObject.t2 > clipLine.t2
+            ? clipLine.t2
+            : interceptionObject.t2;
+      } else {
+        t1 = clipLine.t1;
+        t2 = clipLine.t2;
+      }
```

Comment:
if interserctWithLine returned null instead of undefined, then you could simply write:
```
t1 = Math.max(interceptionObject?.t1, clipLine.t1)
t2 = Math.min(interceptionObject?.t2, clipLine.t2)
```

Reply:
I think no because even if interceptionObject is null, interceptionObject?.t1 will be undefined, not null, leading to t1 = undefined

Reply:
Fair enough, then:

```
t1 = interceptionObject?.t1 > clipLine.t1 ? interceptionObject.t1 : clipLine.t1
t2 = interceptionObject?.t2 < clipLine.t2 ? interceptionObject.t2 : clipLine.t2
```